Sources/SwiftFusion/Inference/NewBetweenFactorAlternative.swift

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+
+/// A factor that specifies a difference between two poses.
+///
+/// `JacobianRows` specifies the `Rows` parameter of the Jacobian of this factor. See the
+/// documentation on `NewJacobianFactor.jacobian` for more information. Use the typealiases below to
+/// avoid specifying this type parameter every time you create an instance.
+public struct NewBetweenFactorAlternative<JacobianRows: FixedSizeArray>:
+  NewLinearizableFactor
+  where JacobianRows.Element == Tuple2<Pose3.TangentVector, Pose3.TangentVector>
+{
+  public typealias Variables = Tuple2<Pose3, Pose3>
+
+  public let edges: Variables.Indices
+  public let difference: Pose3
+
+  public init(_ startId: TypedID<Pose3, Int>, _ endId: TypedID<Pose3, Int>, _ difference: Pose3) {
+    self.edges = Tuple2(startId, endId)
+    self.difference = difference
+  }
+
+  public typealias ErrorVector = Vector12
+  public func errorVector(_ start: Pose3, _ end: Pose3) -> ErrorVector {
+    let actualMotion = between(start, end)
+    let R = actualMotion.coordinate.rot.coordinate.R + (-1) * difference.rot.coordinate.R
```

Comment:
- does not work??

Reply:
Yes, this is a protocol constraint issue